### ISSUE TITLE
Feature Request: use an internal ACME instance

### ISSUE THREAD

**Issue by andreasschulze** [danmarg/sts-mate#2]
the current version is able to fetch certificates from Let's Encrypt. For staging purposes one can specify `--staging`

My usecase is an internal Buildsystem unable to reach let's encrypt but an internal ACME instance.
It would be helpfull to use `--acme-directory-url=https://acme.local/dir`

`--staging` would be only a special case of `--acme-directory-url=https://acme-staging.api.letsencrypt.org/directory`

**Comment by andreasschulze** [comment]
clarification: support for an interal ACME is indented to enable automated testing, not for production ...

**Comment by danmarg** [comment]
This makes sense and looks pretty easy. Though I'd feel more comfortable if this codebase had any sort of unit tests, etc. ;)

Let's see if I remember how to code...

**Comment by danmarg** [comment]
Andreas, can you look at 745ba38838cd2da618690992725924a6d3e47e6f and see if this is what you meant?

**Comment by andreasschulze** [comment]
looks "not so wrong" :-) I'll give it a try next days and report. Just now: Thank you so much...

I agree, unittest would be good, but I can't assist, sorry

**Comment by andreasschulze** [comment]
I built the branch `issue-2`, works fine. Unfortunately I learned, our internal ACME-Instance don't support `tls-alpn-01`.
So all I could test today is `--acme_endpoint=https://acme-v02.api.letsencrypt.org/directory` an this work as expected

My internal feature request to support `tls-alpn-01` wasn't rejected immediately, so I may come back with better results some day.

Unrelated to this issue, I noticed a minor nit:
Parameter are shown as `-help` or `--help`. Both versions work but it looks a little bit inconsistent.
